ddtrace/internal/ci_visibility/writer.py

```diff
@@ -0,0 +1,79 @@
+import os
+from typing import Dict
+from typing import Optional
+
+import ddtrace
+from ddtrace import config
+from ddtrace.vendor.dogstatsd import DogStatsd
+
+from .. import agent
+from .. import service
+from ...sampler import BasePrioritySampler
+from ...sampler import BaseSampler
+from ..runtime import get_runtime_id
+from ..writer import HTTPWriter
+from ..writer import get_writer_interval_seconds
+from .encoder import CIVisibilityEncoderV01
+
+
+class CIVisibilityWriter(HTTPWriter):
+    RETRY_ATTEMPTS = 5
+    HTTP_METHOD = "PUT"
+    STATSD_NAMESPACE = "civisibilitywriter"
+
+    def __init__(
+        self,
+        intake_url,  # type: str
+        sampler=None,  # type: Optional[BaseSampler]
+        priority_sampler=None,  # type: Optional[BasePrioritySampler]
+        processing_interval=get_writer_interval_seconds(),  # type: float
+        timeout=agent.get_trace_agent_timeout(),  # type: float
+        dogstatsd=None,  # type: Optional[DogStatsd]
+        sync_mode=True,  # type: bool
+        report_metrics=False,  # type: bool
+        api_version=None,  # type: Optional[str]
+        reuse_connections=None,  # type: Optional[bool]
+        headers=None,  # type: Optional[Dict[str, str]]
+    ):
+        encoder = CIVisibilityEncoderV01(0, 0)
+        encoder.set_metadata(
+            {
+                "language": "python",
+                "env": config.env,
+                "runtime-id": get_runtime_id(),
+                "library_version": ddtrace.__version__,
+            }
+        )
+
+        headers = headers or dict()
+        headers["dd-api-key"] = os.environ.get("DD_API_KEY", "dummyapikey")
```

Comment:
If we are using this writer (ie: if Agentless is enabled) but `DD_API_KEY` is not set, we want to error out instead of continuing with a dummy key.